src/docs/sphinx/dev_guide/ubuntu_build.rst

```diff
@@ -0,0 +1,192 @@
+.. ## Copyright (c) Lawrence Livermore National Security, LLC and
+.. ## other Serac Project Developers. See the top-level COPYRIGHT file for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+
+.. _ubuntu_build-label:
+
+==================
+Serac Ubuntu Build
+==================
+
+------------------
+Basic System Setup
+------------------
+
+We recommend installing some basic system-level development packages to minimize the
+amount of packages that Spack will build.
+
+Install clang version 14 and make it the default compiler:
+
+.. code-block:: bash
+
+    sudo apt install -y --no-install-recommends clang-14 clang-format-14 llvm-14 libomp-14-dev gfortran-13
+    # Set clang-14 as the default clang
+    sudo update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-14 100 \
+    && sudo update-alternatives --install /usr/bin/clang clang /usr/bin/clang-14 100 \
+    && sudo update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-14 100
+
+Install required build packages to minimize what Spack will build:
+
+.. code-block:: bash
+
+    sudo apt-get -qq install -y --no-install-recommends build-essential bzip2 cmake libopenblas-dev \
+    lua5.2 lua5.2-dev openmpi-bin libopenmpi-dev unzip
+
+Optionally you can install packages to generate documentation:
+
+.. code-block:: bash
+
+    sudo apt-get -qq install -y --no-install-recommends graphviz python3-sphinx texlive-full doxygen
+
+
+-------------------------------
+Generate Spack Environment File
+-------------------------------
+
+Spack uses an environment file to describe where system level packages are to minimize what it builds.
+The following command will download the specific Spack version we use and run a minimal set of commands to
+generate an environment file for you. This should be a good starting point and should be used in following
+Spack builds.
+
+.. code-block:: bash
+
+    scripts/uberenv/uberenv.py --prefix=<path/outside/repository> --setup-and-env-only
```

Comment:
This took forever for me to run.  Is it worth mentioning the environment file that resides in `scripts/spack/configs/linux_ubuntu_24` may be a good enough starting point?  I've been able to use that without much modification in my experience.

Reply:
Or mention the host-configs that you're adding to the repo with this PR

Reply:
The host-config will not work for anyone other than me due to the path to my TPL build that won't exist in their laptop.